zap/src/parser/convert.rs

```diff
@@ -319,17 +321,24 @@ impl<'src> Converter<'src> {
 		let from = evdecl.from;
 		let evty = evdecl.evty;
 		let call = evdecl.call;
-		let data = evdecl
-			.data
-			.as_ref()
-			.map(|types| types.iter().map(|ty| self.ty(ty)).collect::<Vec<_>>());
+		let data = evdecl.data.as_ref().map(|parameters| {
+			parameters
+				.parameters
+				.iter()
+				.map(|(identifier, ty)| {
+					let name = identifier.map(|identifier| identifier.name);
```

Comment:
Are parms with the same name considered okay? If so that would explain my first comment.

Reply:
If you think we need an error I can add it. I didn't because it should be pretty obvious you can't have duplicate parameter names.

Reply:
I think we should add an error, yeah.